jf_agent/bb_download.py

```diff
@@ -88,23 +90,22 @@ def get_default_branch_commits(client, repos):
         api_project = client.projects[repo['project']['login']]
 
         try:
-            commits = api_project.repos[repo['name']].commits(
-                until=repo['default_branch_name'])
+            commits = api_project.repos[repo['name']].commits(until=repo['default_branch_name'])
+
+            for commit in tqdm(
+                commits,
+                desc=f'downloading {repo["project"]["login"]}/{repo["name"]} commits',
+                unit='commit',
+            ):
+                yield _normalize_commit(commit, repo)
```

Comment:
@eliwind to confirm, this is the only non-formatting change?

Reply:
That's right

Reply:
the problem was that we had a try/catch around the thing that returns a generator of commits, but the NotFound wasn't actually thrown until you iterator over them